File: qqbot-module-api/src/main/java/com/mieai/qqbot/module/api/ModuleServiceRegistration.java
package com.mieai.qqbot.module.api;

/** Revocable ownership handle for a service published by a framework module. */
public interface ModuleServiceRegistration extends AutoCloseable {
    String ownerModuleId();
    ModuleServiceKey<?> key();
    boolean isActive();
    @Override void close();
}

File: qqbot-module-api/src/main/kotlin/com/mieai/qqbot/module/api/ModuleServiceRegistration.kt
package com.mieai.qqbot.module.api

/** Revocable ownership handle for a service published by a framework module. */
interface ModuleServiceRegistration : AutoCloseable {
    val ownerModuleId: String

    val key: ModuleServiceKey<*>

    val isActive: Boolean

    override fun close()
}
